data/modules/SearchRescue/SearchRescue.lua

```diff
@@ -1813,6 +1813,10 @@ local onEnterSystem = function (player)
 	 mission.target = createTargetShip(mission)
       end
    end
+	  
+      for _,discarded_ship in pairs(discarded_ships) do
+		table.remove(discarded_ships, discarded_ship)
```

Comment:
I don't think this works. I think you have to do: table.remove(table_to_remove_from, index_of_item) instead of table.remove(table_to_remove_from, item_itself).

Might be faster to just overwrite the whole table with discarded_ships = {}?